parquet/writer.py

```diff
@@ -454,26 +478,28 @@ def write(filename, data, partitions=[0, 500], encoding=parquet_thrift.Encoding.
     open_with: function
         When called with a path/URL, returns an open file-like object
     """
+    sep = sep_from_open(open_with)
     if file_scheme == 'simple':
-        fname = filename
+        fn = filename
     else:
-        def_mkdirs(filename)
-        fname = os.path.join(filename, '_metadata')
-    with open_with(fname) as f:
+        mkdirs(filename)
+        fn = sep.join([filename, '_metadata'])
+    with open_with(fn) as f:
```

Comment:
Thoughts on `with open_with(fn, mode='wb')` ?  This would allow us to drop the `default_open` function for a more generic `open`.